clang/test/CodeGenCXX/debug-info-verbose-trap.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clang_cc1 -triple arm64-apple-ios -std=c++20 -emit-llvm -debug-info-kind=limited %s -o - | FileCheck %s
```

Comment:
Do we want to add a clang test? I thought we generally don't enable optimization in clang CodeGen tests because it makes the tests fragile. Has the guideline changed?

Reply:
Really? We don't test the generated IR in an optimized build? That seems like a bad idea given that code built for production use typically is optimized.

Reply:
Really? We don't test the generated IR in an optimized build? That seems like a bad idea given that code built for production use typically is optimized.

Reply:
@ahatanak Any follow up?

Reply:
If you want to test optimization behavior, that would be an LLVM test starting from IR. Clang itself does not optimize and so should not test optimization behavior.

Reply:
https://llvm.org/docs/TestingGuide.html#best-practices-for-regression-tests recommends we run a minimal set of passes. As @pogo59 said, I think we should add a separate LLVM test if we want to test the merging behavior.

Reply:
@pogo59 @ahatanak Thanks for the explanation.

A slight issue with that is some cases Clang will generate different (unoptimized IR) depending on the optimization level (see `CodeGenFunction::EmitTrapCheck`) so in those cases you have to test Clang's optimizing behavior. You're not using that here so that's fine

Reply:
In that case, we can use `-disable-llvm-passes` or `-disable-llvm-optzns` to avoid running the llvm optimization passes.